giantswarm-platform-labels/README.md

```diff
@@ -0,0 +1,54 @@
+---
+creation_date: 2025-02-25
+issues: []
+owners:
+- https://github.com/orgs/giantswarm/teams/sig-architecture
+state: review
+summary: XXXXX
+---
+
+# Title
+Standardizing Platform Component Labeling
+
+### Problem statement
+<!-- Explain the issue or challenge that needs to be addressed. This should include background information and context to help stakeholders understand why this decision is important. -->
+Currently, platform components are not labeled to indicate their belonging to specific subsystems. This lack of a standardized subsystem label creates some operational challenges.
+
+Firstly, subsystems composed of multiple applications lack a unified identifier. While individual applications are labeled, targeting an entire subsystem requires listing each application's specific label, leading to cumbersome and error-prone configurations. For example, network policies for DNS traffic must explicitly target both coredns and k8s-dns-node-cache, exposing the implementation details to customers.
+
+Secondly, changes in the underlying implementation of subsystems requires widespread label updates. When the prometheus-agent was migrated to alloy, the labels used for metric and log scraping had to be updated across various tools and documentation, requiring a transitional period where both old and new labels were maintained. This introduces unnecessary complexity and maintenance overhead.
+
+The absence of a generic subsystem label hinders our ability to target pods in a flexible and maintainable manner in NetworkPolicies or other operational tooling. This lack of abstraction exposes implementation details, increases maintenance burden, and complicates operational tasks. We have identified at least two key use cases where this issue is prominent: DNS traffic management and metric/log scraping during implementation changes. This issue impacts multiple applications and subsystems, highlighting the need for a standardized approach to subsystem labeling.
+
+### Decision maker
+<!-- Identify the person (preferred) or a group responsible for making the final decision. -->
+SIG-Architecture
+
+### Who is affected / stakeholders
+<!-- List the individuals, teams, or SIGs that will be impacted by this decision and must provide feedback. -->
+
+### Preferred solution
+<!-- Describe the solution that is currently favored based on the analysis of the problem. -->
+
+This RFC proposes introducing a label, `giantswarm.io/subsystem`, to identify pods belonging to specific platform subsystems. This label will enable us to target groups of pods based on their subsystem affiliation, simplifying operations and improving maintainability.
```

Comment:
`platform.giantswarm.io/subsystem` or `platform.giantswarm.io/component`?

Reply:
One vote for `subsystem`. It's generic enough that we can probably (ab)use it in several ways, and `component` seems too granular for the listed purposes.
```suggestion
This RFC proposes introducing a label, `platform.giantswarm.io/subsystem`, to identify pods belonging to specific platform subsystems. This label will enable us to target groups of pods based on their subsystem affiliation, simplifying operations and improving maintainability.
```